### DISCUSSION TITLE
CGAL fails to detect some non-manifold vertices

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#7160] [Q&A]
Hello,

According to the R package **Rvcg**, the mesh in the attached **off** file has two non-manifold vertices. However CGAL does not find them. I proceed as follows:

```cpp
    std::vector<halfedge_descriptor> hds;
    std::back_insert_iterator<std::vector<halfedge_descriptor>> bii = std::back_inserter(hds);
    PMP::non_manifold_vertices(mymesh, bii);
    Rcpp::Rcout << "NMV: " << hds.size() << "\n";
```

Is it correct? 

The computation of the normals crashes the session, but I think it would work if these two vertices were removed (because this is the behavior in R).

[clippedmesh.off.txt](https://github.com/CGAL/cgal/files/10342154/clippedmesh.off.txt)

**Comment by stla** [comment]
Moreover `CGAL::is_closed()` claims that this mesh is closed:

![Capture du 2023-01-04 09-48-00](https://user-images.githubusercontent.com/4466543/210517598-fb951730-dcc8-4f35-82e7-88d99fd541d2.png)

**Comment by afabri** [comment]
When I load the off file in the Polyhedron demo of CGAL I get the warning that there are 472 isolated vertices and that non-manifold vertices have been found.   Can you please put a self contained cpp on gist.github.com instead of a 4 lines of code in a text. 
And the stistics are:
![image](https://user-images.githubusercontent.com/3263539/210532901-57c93569-1a49-45b2-b949-2b3309074e36.png)

**Comment by stla** [comment]
Ah ok, maybe writing the mesh to a file and then reading it solves the issue.

In fact I had an original mesh that I copied and I removed some faces of the copy. That's how I obtained this mesh. Running `PMP::remove_isolated_vertices(mesh)` did not remove anything.

I changed my strategy: now I use

```cpp
PMP::orient_polygon_soup(points, polygons, CGAL::parameters::visitor(soupvis))
```
where `points` are all the vertices of the original mesh, `polygons` are the faces I want to keep, and the visitor is here to print a message if non-manifold vertices are found. This works fine. Then I use `PMP::polygon_soup_to_polygon_mesh`.
